### ISSUE TITLE
Add dictionary based segmenter

### ISSUE THREAD

**Issue by makotokato** [unicode-org/icu4x#1033]
For https://github.com/unicode-org/icu4x/issues/813 as CJ

**Comment by makotokato** [comment]
Also, this is prototype. https://github.com/makotokato/dictionary_segmenter

**Comment by aethanyc** [comment]
This is part of #943 and can be used in existing line breaker.

**Comment by sffc** [comment]
For Lao, Khmer, Thai, and Burmese, we could use LSTM or we could use Dictionary or we could have a way to toggle between them.  We should have a discussion about which approach we take here.

**Comment by sffc** [comment]
@aethanyc to help find someone to help LSTM be production-ready.

**Comment by makotokato** [comment]
@aethanyc
Should I add new crate (`segmenter_dictionary`) in experimental, instead of adding this to segmenter crate?

**Comment by sffc** [comment]
I think we should eventually be moving the LSTM into the segmenter crate, and leverage Cargo features where necessary to turn off dependencies.  I do not think there is a lot of value in having a separate segmenter_lstm or segmenter_dictionary crate.

**Comment by aethanyc** [comment]
I agree with Shane. Let's implement dictionary segmenter directly in `segmenter` crate. (Merging `segmenter_lstm` into `segmemter` can get rid of external dependency like #1531)

**Comment by makotokato** [comment]
Actually, we use dictionary simply now, but some Burmese test cases will be failure since combine character isn't first (SpacingMark etc in UAX29's Grapheme segmenter).

Ex,  ICU's Burmese dictionary has U+0x1005 U+0x102c U+0x1019 as one word, but U+0x1005 U+0x102c U+0x1019 U+0x103c are 2 words (U+0x1005 U+0x102c and U+0x1019 U+0x103c)

So I am investigate to fix it.
